### PR TITLE
Fix Panic: When odo app describe run out of component folder

### LINKED ISSUE #2528: odo application describe crash when run out of a component folder
/kind bug


## What versions of software are you using?

**Operating System:**
Win10Pro

**Output of `odo version`:**
```
odo v1.0.3 (HEAD)

Server: https://172.18.1.78:8443
Kubernetes: v1.11.0+d4cacc0
```

## How did you run odo exactly?

`odo application describe myapp --project myproj`

## Actual behavior
```
Application Name: sbapp has 1 component(s) and 1 service(s):                                                                                               
--------------------------------------                                                                                                                     
Component Name: sbcomp1                                                                                                                                    
Type: java                                                                                                                                                 
Source: file:///./                                                                                                                                         
Environment Variables:                                                                                                                                     
 · DEBUG_PORT=5858                                                                                                                                         
panic: runtime error: slice bounds out of range                                                                                                            
                                                                                                                                                           
goroutine 1 [running]:                                                                                                                                     
github.com/openshift/odo/pkg/odo/util.PrintComponentInfo(0xc00095edc0, 0xc00057bc58, 0x7, 0x19d4618, 0x9, 0x19e9cb5, 0x19, 0xc00057bc58, 0x7, 0x0, ...)    
        /builddir/build/BUILD/gocode/src/github.com/openshift/odo/pkg/odo/util/cmdutils.go:116 +0x13b7                                                     
github.com/openshift/odo/pkg/odo/cli/application.(*DescribeOptions).Run(0xc00084f230, 0x0, 0x0)                                                            
        /builddir/build/BUILD/gocode/src/github.com/openshift/odo/pkg/odo/cli/application/describe.go:88 +0x4ec                                            
github.com/openshift/odo/pkg/odo/genericclioptions.GenericRun(0x1ca18e0, 0xc00084f230, 0xc0007e4480, 0xc000990030, 0x1, 0x3)                               
        /builddir/build/BUILD/gocode/src/github.com/openshift/odo/pkg/odo/genericclioptions/runnable.go:28 +0x107                                          
github.com/openshift/odo/pkg/odo/cli/application.NewCmdDescribe.func1(0xc0007e4480, 0xc000990030, 0x1, 0x3)                                                
        /builddir/build/BUILD/gocode/src/github.com/openshift/odo/pkg/odo/cli/application/describe.go:117 +0x65                                            
github.com/openshift/odo/vendor/github.com/spf13/cobra.(*Command).execute(0xc0007e4480, 0xc000990000, 0x3, 0x3, 0xc0007e4480, 0xc000990000)                
        /builddir/build/BUILD/gocode/src/github.com/openshift/odo/vendor/github.com/spf13/cobra/command.go:702 +0x290                                      
github.com/openshift/odo/vendor/github.com/spf13/cobra.(*Command).ExecuteC(0xc0007ed200, 0x1ad86e0, 0xc000644d80, 0x0)                                     
        /builddir/build/BUILD/gocode/src/github.com/openshift/odo/vendor/github.com/spf13/cobra/command.go:783 +0x2d1                                      
github.com/openshift/odo/vendor/github.com/spf13/cobra.(*Command).Execute(...)                                                                             
        /builddir/build/BUILD/gocode/src/github.com/openshift/odo/vendor/github.com/spf13/cobra/command.go:736                                             
main.main()                                                                                                                                                
        /builddir/build/BUILD/gocode/src/github.com/openshift/odo/cmd/odo/odo.go:72 +0x3f6                                                                 
```

## Expected behavior

Should not crash

## Any logs, error output, etc?

### PR DESCRIPTION
**Pull request by adisky**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespace from that line:

> /kind api-change
/kind bug
> /kind cleanup
> /kind deprecation
> /kind design
> /kind documentation
> /kind failing-test
> /kind feature
> /kind flake
> /kind code-refactoring

**What does does this PR do / why we need it**:
 odo app describe command does not work, When run out of component folder, it lists storage from local config. This PR updates it to list storage from oc client, It also adds bounds check at other places to avoid panic in case of no output.

**Which issue(s) this PR fixes**: 

Fixes #2528

**How to test changes / Special notes to the reviewer**:
Run `odo application describe <app> --project <myproject>` after adding storage to a component, from out of component folder, it should work now.

### CODE REVIEW

File: pkg/odo/util/cmdutils.go [hunk 1]
```diff
@@ -95,26 +91,32 @@ func PrintComponentInfo(client *occlient.Client, currentComponentName string, co
...
+		}
+
 	}
 
 	// Storage
 	if len(componentDesc.Spec.Storage) > 0 {
 
 		// Retrieve the storage list
-		storages, err := localConfig.StorageList()
+		storages, err := storage.List(client, currentComponentName, applicationName)
```
Comment (adisky) [inline pkg/odo/util/cmdutils.go]: @kadel @girishramnani @mik-dass So as per the discussions, should i remove this change, and only keep bounds check here? and the `state` part will be implemented as a part of feature request?

**Comment by kadel** [comment]
@adisky please add at least integration test to verify that the bug is fixed, and to prevent reintroducing the bug again in the future

**Comment by girishramnani** [comment]
also please add WIP if there are tests to be written for this @adisky

**Comment by adisky** [comment]
@kadel @girishramnani I have added an Integration Test.

**Pull request merged**